user and role related tables

TengLiu306750 opened the pull request:
# UPEI Behavior Rating Tool PR

Before creating a pull request, ensure you are not committing broken code and if you have trouble with any particular problem, do not hesitate to contact another group member for assistance.

## What are the contents of this PR?

- [x] Refactor
- [x] New Feature
- [ ] Optimization
- [ ] Documentation
- [x] Bug Fix
- [ ] Configuration
- [ ] Breaking Change

## Describe the PR

Added user-role-permission model. there is an "info" (type is JSON) column for almost all the tables which can be used when adding more attributes wither alter the tables. 

## Related issue

The feature:  3-26 Create user/admin account models 

## Checklist

- [x ] My code follows the same style of the project
- [ ] My code requires an update to the documentation
- [ ] I have updated the documentation as required
- [ ] I have added the required unit/feature tests
- [ ] All tests were successful

## Testing

Besides automated testing, please describe how you tested your changes below:

zmclure committed:
Create simple .csv upload system

Can be accessed and tested by visiting "localhost:8080/upload".  The next step is to extract information from string array so that it may be sent to the database.

TengLiu306750 committed:
user and role related tables

zmclure committed:
Create method to upload individual user

aisenor committed:
Fields to add single user details, buttons for multiple user import

aisenor committed:
Added example image to add_user

zmclure committed:
Merge branch 'feature/db-user-model' into feature/upload-csv

zmclure commented on src/public/sql/initial.sql:
```diff
@@ -0,0 +1,108 @@
+    first_name VARCHAR(20), 
+    middle_name VARCHAR(20),
+    last_name VARCHAR(20), 
+    email VARCHAR(20),
```
We should probably use VARCHAR(255) to account for some longer emails

TengLiu306750 replied:
Yes, right, I will change that. Thanks for point out.

aisenor replied:
Can we get rid of username and middle name, and possibly ID if it isn't being used anywhere else? Email could be our unique identifier.

zmclure committed:
Create function to upload new user to DB

aisenor committed:
Added image

aisenor committed:
Added last name field

aisenor committed:
Added template download and photo

zmclure committed:
Merge branch 'feature/upload-csv' into feature/add-user-page

zmclure committed:
Set default new user password to be "password"

Temporary fix until we start hashing passwords for new accounts

zmclure committed:
Upload users from csv

zmclure committed:
Delete /upload page that was used for csv upload tests

zmclure committed:
Update UploadController.php

zmclure committed:
Fix to make it return to the current page after submitting a form

aisenor committed:
Bug comments

zmclure committed:
Update UploadController.php

Removed the ['id' => 1] part of the route redirect.  What this did was add a "?id=1" at the end of the URL.

zmclure committed:
Changed action for adding multiple users from file

zmclure committed:
Fixed no file upload issue

Trying to upload with no file used to take you to a blank page

zmclure committed:
Allow uploading of single user to database

zmclure committed:
Update UploadController.php

Added comments on functions.

aisenor committed:
Removed comments

zmclure committed:
Display message notifying how many new users were added in file upload

still needs some formatting

zmclure committed:
Display confirmation for uploading single new user

TengLiu306750 committed:
user and role related tables

teng-liu committed:
laravel users-roles-permissions

teng-liu committed:
added testing data, not done yet

teng-liu committed:
firstName/lastName to first_name, last_name

zmclure committed:
firstName/lastName to first_name/last_name

zmclure committed:
Merge branch 'feature/db-user-model' into feature/add-user-page

zmclure committed:
Register newly added users and has default password

users created through the /add_user page are now able to log in to the site

zmclure committed:
Check for duplicate emails

aisenor committed:
Added account page template

aisenor committed:
First and Last name placeholders

aisenor committed:
Account page frontend added

zmclure committed:
Account page can change current user's name and email

zmclure committed:
User now able to change their password, added confirmation/error messages

aisenor committed:
Added to-do message, program crashes when trying to update user info

aisenor committed:
Changed style of text to match other pages

aisenor committed:
Fixed response message format

aisenor committed:
Added email placeholder to be current users email

aisenor committed:
Updated response message formatting

zmclure committed:
Assign roles to new users in db table

"roles" and "users_roles" are now updated whenever a new user is created

chrismacdonaldw committed:
Updated UserFactory

During review, noticed that the UserFactory should have been updated to support changes made to the user model.

chrismacdonaldw committed:
Removed redundant code

zmclure committed:
create empty json object upon user creation

zmclure committed:
Update UploadController.php

aisenor committed:
Form will not submit unless old email matches the users account email

zmclure committed:
changed alert type for specific conditions on file upload

all users = success
some users = warning
no users = danger

zmclure committed:
added email format validation for uploading files

aisenor committed:
Display first and last name in the task bar

aisenor committed:
Merge branch 'feature/add-user-page' of https://github.com/CS48X0-Behavior-Tool-Project/behavior-rating-tool into feature/add-user-page

aisenor committed:
This page is done

teng-liu committed:
Merge branch 'feature/add-user-page' into feature/db-user-model

teng-liu committed:
user-role model done, added role in the task bar

TengLiu306750 commented:
User-Role_Permission model is done!

TengLiu306750 commented:
User-Role_Permission model is done! Please review and merge.

aisenor committed:
Removed role from toolbar moved to home page
It doesn't look good in the toolbar, if we just need it for testing 
logins then it has been moved to the landing page

aisenor committed:
replaced auth()-> with Auth::
They mean the exact same thing but we should be consistent throughout 
our code. I believe in the routing file it has to be Auth:: so i changed 
all others to match

aisenor commented on src/app/Http/Controllers/PermissionController.php:
```diff
@@ -0,0 +1,74 @@
+
+class PermissionController extends Controller
+{   
+	// create system level data for:  Role, User, Permissions 
```
Should this include TA and Student roles also? It only includes admin and experts

aisenor committed:
Update 2021_02_07_071112_create_roles_permissions_table.php

Fixed indentation

aisenor commented on src/public/sql/data.sql:
```diff
@@ -0,0 +1,39 @@
+
+INSERT IGNORE INTO roles (id, name, slug)
+ VALUES
+    (1, 'administrator', 'administrator'),
```
We need to include TA in here, they should have the ability to do everything except delete (which hasn't even been made yet)

aisenor reviewed (approved):
Just need permissions for a TA but they are essentially an Admin that shouldn't be able to delete things and right now, no one can delete anything

aisenor committed:
Merge branch 'main' into feature/db-user-model

aisenor committed:
replaced auth()-> with Auth::

chrismacdonaldw commented:
Tests are failing because we are manually creating a role table when the Bouncer migrations already creates that for us.
